printing struct with thread produces C error

### Describe the bug

Printing a struct with a field of type `thread string` generates a C error: cannot convert 'struct <anonymous>' to 'int'

### Reproduction Steps

```v
// Video is the object containing all the information needed for
// conversion to ascii.
struct Video {
pub:
	// path may be URL or path on local file system.
	path         string
	tmp_dir      string
	frame_count  int
	frame_format FrameFormat
pub mut:
	// TODO: once V fixes bug #19281 replace the type here
	// with `thread !`. And the return type on the corresponding
	// function that is spawned.
	extract_frames_thread thread string
}

enum FrameFormat {
	bmp
	jpg
}

fn (video Video) extract_frames() string {
	// try to extract frames os.execute()
	// if os.execute() returns with an exit
	// code other than 0, return an error.
	if true {
		return 'Failed to extract frames: os.execute() cmd output'
	}
	// do something else
	return ''
}

fn main() {
	mut video := Video{
		path: '/path/to/some/mp4'
		tmp_dir: '/tmp'
	}
	video.extract_frames_thread = spawn video.extract_frames()
	println(video)
}
```

### Expected Behavior

Expected code to compile and print the struct.

### Current Behavior

Generates a C error.
```
==================
C:/Users/imado/AppData/Local/Temp/v_0/test.10842827188560893074.tmp.c:500: warning: WINVER redefined
C:/Users/imado/AppData/Local/Temp/v_0/test.10842827188560893074.tmp.c:2198: error: cannot convert 'struct <anonymous>' to 'int'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V full version: V 0.4.1 639f128

### Environment details (OS name and version, etc.)

V full version: V 0.4.1 639f128
OS: windows, Microsoft Windows 11 Pro v22621 64-bit
Processor: 16 cpus, 64bit, little endian, 

getwd: C:\Users\imado\Documents\test
vexe: C:\Users\imado\v\v.exe
vexe mtime: 2023-09-05 17:00:49

vroot: OK, value: C:\Users\imado\v
VMODULES: OK, value: C:\Users\imado\.vmodules
VTMP: OK, value: C:\Users\imado\AppData\Local\Temp\v_0

Git version: git version 2.33.1.windows.1
Git vroot status: 0.4.1-10-g639f128c
.git/config present: true

CC version: Error: 'cc' is not recognized as an internal or external command,
operable program or batch file.

thirdparty/tcc status: thirdparty-windows-amd64 e90c2620

--- a/vlib/v/gen/c/auto_str_methods.v
+++ b/vlib/v/gen/c/auto_str_methods.v
@@ -846,7 +846,7 @@ fn (g &Gen) type_to_fmt(typ ast.Type) StrIntpType {
 	if typ.is_int_valptr() || typ.is_float_valptr() {
 		return .si_s
 	} else if sym.kind in [.struct_, .array, .array_fixed, .map, .bool, .enum_, .interface_,
-		.sum_type, .function, .alias, .chan] {
+		.sum_type, .function, .alias, .chan, .thread] {
 		return .si_s
 	} else if sym.kind == .string {
 		return .si_s
@@ -1082,6 +1082,8 @@ fn struct_auto_str_func(sym &ast.TypeSymbol, lang ast.Language, _field_type ast.
 		return '${fn_name}(${obj})', true
 	} else if sym.kind == .chan {
 		return '${fn_name}(${deref}it.${final_field_name}${sufix})', true
+	} else if sym.kind == .thread {
+		return '${fn_name}(${deref}it.${final_field_name}${sufix})', false
 	} else {
 		mut method_str := ''
 		if !field_type.is_ptr() && (field_type.has_flag(.option) || field_type.has_flag(.result)) {

--- a/vlib/v/slow_tests/inout/printing_struct_with_thread_field.out
+++ b/vlib/v/slow_tests/inout/printing_struct_with_thread_field.out
@@ -1,0 +1,7 @@
+Video{
+    path: '/path/to/some/mp4'
+    tmp_dir: '/tmp'
+    frame_count: 0
+    frame_format: bmp
+    extract_frames_thread: thread(string)
+}

--- a/vlib/v/slow_tests/inout/printing_struct_with_thread_field.vv
+++ b/vlib/v/slow_tests/inout/printing_struct_with_thread_field.vv
@@ -1,0 +1,40 @@
+// Video is the object containing all the information needed for
+// conversion to ascii.
+struct Video {
+pub:
+	// path may be URL or path on local file system.
+	path         string
+	tmp_dir      string
+	frame_count  int
+	frame_format FrameFormat
+pub mut:
+	// TODO: once V fixes bug #19281 replace the type here
+	// with `thread !`. And the return type on the corresponding
+	// function that is spawned.
+	extract_frames_thread thread string
+}
+
+enum FrameFormat {
+	bmp
+	jpg
+}
+
+fn (video Video) extract_frames() string {
+	// try to extract frames os.execute()
+	// if os.execute() returns with an exit
+	// code other than 0, return an error.
+	if true {
+		return 'Failed to extract frames: os.execute() cmd output'
+	}
+	// do something else
+	return ''
+}
+
+fn main() {
+	mut video := Video{
+		path: '/path/to/some/mp4'
+		tmp_dir: '/tmp'
+	}
+	video.extract_frames_thread = spawn video.extract_frames()
+	println(video)
+}

